src/map/ai/controllers/trust_controller.cpp

```diff
@@ -45,10 +46,26 @@ CTrustController::~CTrustController()
     POwner->allegiance = ALLEGIANCE_PLAYER;
 }
 
+void CTrustController::Despawn()
+{
+    if (POwner)
+    {
+        POwner->PMaster = nullptr;
+    }
+    CController::Despawn();
+}
+
 void CTrustController::Tick(time_point tick)
 {
     m_Tick = tick;
 
+    if (POwner->PAI->IsCurrentState<CDespawnState>())
```

Comment:
should just do what mob controller does instead of needing to check specifically despawn state (death state probably also does it, along with any other state we may ever add)

instead of this check, just add a check for `!PMob->isDead` to the else before RoamTick

Reply:
Done. https://github.com/DarkstarProject/darkstar/pull/4739/commits/26bbbf89557ed574561cdbb1015c86119637499b